config_builder.py

```diff
@@ -0,0 +1,251 @@
+                _kregex = getenv('KUBERNETES_NAMESPACE_NAME_REGEX')
+                if _kregex is not None:
+                    self.set_instance_property('kubernetes', 'namespace_name_regexp', _kregex)
+
+        if getenv('MESOS_MASTER') is not None:
+            self.set_instance_property('mesos_master', 'url', 'http://leader.mesos:5050')
+            self.set_instance_property('zk', 'host', 'leader.mesos')
+
+        if getenv('MESOS_SLAVE') is not None:
+            self.set_instance_property('mesos_slave', 'url', 'http://{}:5051'.format(getenv('HOST', 'localhost')))
+
+        if getenv('MARATHON_URL') is not None:
+            _murl = getenv('MARATHON_URL')
+            if _murl is not None:
+                self.set_instance_property('marathon', 'url', _murl)
+
+    def load_config(self, config_file):
+        '''
+        Loads a config file using ConfigParser
+        '''
+        self.config = ConfigParser()
+        # import existing config from file
+        with open(config_file, 'rb') as cfd:
+            self.config.readfp(cfd)
+
+    def save_config(self, config_file):
+        '''
+        Saves a ConfigParser object (self.config) to the given file
+        '''
+        if self.config is None:
+            logging.error('config object needs to be created before saving anything')
+            exit(1)
+        with open(config_file, 'wb') as cfd:
+            self.config.write(cfd)
+
+    def build_supervisor_conf(self):
+        '''
+        Builds the supervisor.conf based on the environment variables
+        '''
+        self.load_config(self.supervisor_conf_file)
+
+        _logs_stdout = getenv('DD_LOGS_STDOUT', getenv('LOGS_STDOUT', 'no'))
+        for _section in self.config.sections():
+            if getenv('DD_SUPERVISOR_DELETE_USER') is not None and self.config.has_option(_section, 'user'):
+                self.config.remove_option(_section, 'user')
+            if _logs_stdout == 'yes':
+                for _opt in self.config.options(_section):
+                    if _opt.endswith('_logfile'):
+                        self.config.remove_option(_section, _opt)
+                    if _section.startswith('program:'):
+                        self.set_property('stdout_logfile', '/dev/stdout', _section)
+                        self.set_property('stdout_logfile_maxbytes', '0', _section)
+                        self.set_property('stderr_logfile', '/dev/stderr', _section)
+                        self.set_property('stderr_logfile_maxbytes', '0', _section)
+
+        if getenv('KUBERNETES') is not None or getenv('MESOS_MASTER') is not None or getenv('MESOS_SLAVE') is not None:
+            # expose supervisord as a health check
+            if not self.config.has_section('inet_http_server'):
+                self.config.add_section('inet_http_server')
+            self.set_property('port', '0.0.0.0:9001', 'inet_http_server')
+
+        self.save_config(self.supervisor_conf_file)
+
+    def build_datadog_conf(self):
```

Comment:
For datadog.conf I agree it's a good idea to allow individual overrides, and binding all settings to env var is a good way to do that (that's actually what agent 6 will do).